specs/common.yaml

```diff
@@ -1,39 +1,163 @@
-schemas:
-  ErrorMessage:
-    type: object
+parameters:
+  pollTimeout:
+    name: pollTimeout
     description: >
-      Generic Error Message structure.
-    properties:
-      message:
-        type: string
+        Timeout used in long-polling calls (in seconds).
+        How many seconds server should wait for new events (0.0 means it should
+        return immediately if there are no events)
+    required: false
+    in: query
+    schema:
+      type: number
+      format: float
+      minimum: 0.0
+
+  ackTimeout:
+    name: ackTimeout
+    description: >
+        Timeout used in blocking calls waiting for eg. acknowledgement.
+        How many seconds server should wait for response/acknowledgement of an action
+        (0.0 means it should wait for other party's response indefinitely)
+    required: false
+    in: query
+    schema:
+      type: number
+      format: float
+      minimum: 0.0
+
+  maxItems:
+    name: maxItems
+    description: >
+        Maximum number of items that server should return at once
+        (empty value means no limit).
+    required: false
+    in: query
+    schema:
+      type: integer
+      format: int32
+      minimum: 0
 
-parameters:
   maxEvents:
     name: maxEvents
     description: >
-      Maximum number of events that server should return at once
-      (empty value means no limit).
+        Maximum number of events that server should return at once
+        (empty value means no limit).
     required: false
     in: query
     schema:
       type: integer
       format: int32
       minimum: 0
 
-  timeout:
-    name: timeout
-    description: >
-      How many seconds server should wait for new events (0.0 means it should
-      return immediately if there are no events)
+  afterTimestamp:
+    name: afterTimestamp
+    description: Apply only to events later than specified timestamp
+    in: query
     required: false
+    schema:
+      type: string
+      format: date-time
+
+  appSessionId:
+    name: appSessionId
+    description: A correlation/session identifier used for querying events related to an action
```

Comment:
and here
```suggestion
    description: A correlation/session identifier used for querying endpoints related to an action
```

Reply:
Actually the appSessionId is used in endpoints dedicated to querying events, so the description is valid.

Reply:
Agreed.